aten/src/ATen/function_wrapper.py

```diff
@@ -201,19 +217,39 @@ def TypedDict(name, attrs, total=True):  # type: ignore
 """)
 
 # special method definition for factory functions in Functions.h that initializes backends
-C10_FACTORY_DEFINITION = CodeTemplate("""\
-static inline ${return_type} ${api_name}(${formals}) {
+C10_FACTORY_DEFINITION_FACTORY = CodeTemplate("""\
+static inline ${return_type} _${api_name}(${formals}) {
 #ifdef USE_STATIC_DISPATCH
     ${static_dispatch_function_body}
 #else
-    globalLegacyTypeDispatch().initForDispatchKeySet(${inferred_key_set});
     static c10::OperatorHandle op = c10::Dispatcher::singleton()
         .findSchemaOrThrow("aten::${operator_name}", "${overload_name}");
     return op.callUnboxed<${formals_types_with_return}>(${native_actuals});
 #endif
 }
 """)
 
+COLLAPSED_FACTORY_DEFINITION = CodeTemplate("""\
+inline ${return_type} ${api_name}(${collapsed_formals}) {
+    return _${api_name}(${expanded_native_actuals});
+}
+""")
+
+COLLAPSED_METHOD_DEFINITION = CodeTemplate("""\
```

Comment:
Wouldn't it be more direct and easier to just call this `TENSOROPTIONS_METHOD_DEFINITION`